src/plugins/discourse/fetch.js

```diff
@@ -164,6 +168,57 @@ export class Fetcher implements Discourse {
     return this._fetchImplementation(fullUrl, fetchOptions);
   }
 
+  async categoryDefinitionTopicIds(): Promise<Set<TopicId>> {
+    const topicIdRE = new RegExp("/t/[\\w-]+/(\\d+)$");
+    const urls: string[] = [];
+    const categoriesWithSubcategories: CategoryId[] = [];
+
+    // Root categories
+    const response = await this._fetch(
+      `/categories.json?show_subcategory_list=true`
+    );
+    failIfMissing(response);
+    failForNotOk(response);
+    const {categories: rootCategories} = (await response.json()).category_list;
+    for (const cat of rootCategories) {
+      if (cat.topic_url != null) {
+        urls.push(cat.topic_url);
+      }
+      if (cat.subcategory_ids) {
+        categoriesWithSubcategories.push(cat.id);
+      }
+    }
+
+    // Subcategories
+    for (const rootCatId of categoriesWithSubcategories) {
+      const subResponse = await this._fetch(
+        `/categories.json?show_subcategory_list=true&parent_category_id=${rootCatId}`
+      );
+      failIfMissing(subResponse);
+      failForNotOk(subResponse);
+      const {categories: subCategories} = (
+        await subResponse.json()
+      ).category_list;
+      for (const cat of subCategories) {
+        if (cat.topic_url != null) {
+          urls.push(cat.topic_url);
+        }
+      }
+    }
+
+    const ids = urls.map((url) => {
+      const match = topicIdRE.exec(url);
+      if (match == null) {
+        throw new Error(
+          `Encountered topic URL we failed to parse it's TopicId from: ${url}`
+        );
+      }
+      return Number(match[1]);
+    });
+
+    return new Set(ids);
```

Comment:
This is modified from the comment https://github.com/sourcecred/sourcecred/pull/1452#discussion_r346069716


> Why would they not be unique? Can multiple categories share a topic?

---

> Fair point. I've included it as a better safe than sorry, explicit semantic constraint. For efficiency, the syncing relies on those semantics. Perhaps it's better to just change the return type to be a Set<TopicId>.
>
> In terms of the API, I think it shouldn't currently be possible to share topics between categories. But I've not looked too far into it, as it's a pretty niche thing to want very elaborate setups for the "about x category" definition topics.

---

This now has the `Set<TopicId>` type.